src/core/control/PdfCache.h

```diff
@@ -53,12 +54,16 @@ class PdfCache {
      */
     void setRefreshThreshold(double percentDifference);
 
+    static void renderMissingPdfPage(cairo_t* cr, double pageWidth, double pageHeight);
+
 private:
     void setZoom(double zoom);
-    PdfCacheEntry* lookup(const XojPdfPageSPtr& popplerPage);
+    PdfCacheEntry* lookup(size_t pageNo) const;
     PdfCacheEntry* cache(XojPdfPageSPtr popplerPage, cairo_surface_t* img, double zoom);
 
 private:
+    XojPdfDocument pdfDocument;
```

Comment:
Is this meant to be owned by PdfCache or is this supposed to be a reference?

Reply:
XojPdfDocument is itself some kind of smart pointer (It's pretty unreadable and would deserve a do-over, but that's out of scope). So I do want to own one of them, to make sure the pdf document does not get destroyed before the cache does. Both are eventually destroyed when the document is replaced by a newly opened one.